src/main/java/store/reduct/client/ReductClient.java

```diff
@@ -92,14 +98,15 @@ public Bucket createBucket(String name, BucketSettings bucketSettings)
 																									// settings as null
 																									// until invoke
 																									// read.
-		if (httpResponse.statusCode() == 200) {
+		if (httpResponse.statusCode() == 200 || Boolean.TRUE.equals(bucketSettings.getExists())) {
```

Comment:
You pass all errors here including 401 and 403. You need to use the getExists() flag with the 409 error.

Boolean.TRUE.equals(bucketSettings.getExists())) is this really needed!?

```java
if (httpResponse.statusCode() == 200 || ( bucketSettings.getExists() && httpResponse.statusCode() == 409)

```

Reply:
Fixed